plugin/pkg/auth/authenticator/token/bootstrap/bootstrap.go

```diff
@@ -184,3 +192,28 @@ func parseToken(s string) (string, string, error) {
 	}
 	return split[1], split[2], nil
 }
+
+// getGroups loads and validates the bootstrapapi.BootstrapTokenGroupsKey key
+// from the bootstrap token secret, returning a list of group names or an error
+// if any of the group names are invalid.
+func getGroups(secret *api.Secret) ([]string, error) {
+	groupsString := getSecretString(secret, bootstrapapi.BootstrapTokenGroupsKey)
+	if groupsString == "" {
+		return []string{bootstrapapi.BootstrapDefaultGroup}, nil
+	}
+
+	// split on commas amd validate the prefix on each group name
+	groups := strings.Split(groupsString, ",")
+	for _, group := range groups {
+		if group == bootstrapapi.BootstrapDefaultGroup {
+			// allow the default group even though it doesn't strictly match the prefix
+			continue
```

Comment:
I did this sort of halfway. I made it so `system:bootstrappers` is always added for bootstrap tokens, but I still made it so the serialized values in the `auth-extra-groups` key includes the full prefix, so if you set the key to `"system:bootstrappers:foo,system:bootstrappers:bar"` you get three groups:
 - `system:bootstrappers`
 - `system:bootstrappers:foo`
 - `system:bootstrappers:bar`

@ericchiang does that work for you? I prefer for the serialized value to be very explicit so there's just validation but not transformation happening in the authenticator.